c7n/resources/appmesh.py

```diff
@@ -122,7 +122,7 @@ def _describe(self, mesh_and_child_names):
         return results
 
 
-@resources.register('appmesh-virtual-gateway')
+@resources.register('appmesh-virtualgateway')

```

Comment:
Since this resource type name went out in c7n 0.9.35.0, we can't just rename this. But your point about consistency is a good one. We can prefer the consistent name, and leave the previous name as an alias for backward compatibility. This pattern shows up [elsewhere](https://github.com/cloud-custodian/cloud-custodian/blob/2a7a34ad4271a46594fdbd742be8446d8b790c53/tools/c7n_azure/c7n_azure/resources/key_vault_keys.py#L20-L21).

```suggestion
@resources.register('appmesh-virtualgateway', aliases=['appmesh-virtual-gateway'])
```

Reply:
Works for me thanks